Added known limitation section to the docs (#796)
pytest-dev/pytest-xdist: docs/index.rst

    how-to
    how-it-works

>     changelog

Reviewer: The `796.doc` file is being listed as binary, and missing the `.rst` extension, can you please take a look?

Author: Yes i folowed example of 791.doc that is a word file so i did mine as well.... i think this realy needs  contributing section  But i will change it to rst file  definetly

Reviewer: Ahh I see the confusion... actually `791.doc` is a text file:

https://github.com/pytest-dev/pytest-xdist/blob/master/changelog/791.doc

But we missed that during merge, as usually we recommend users to use `.rst` as extension, so that should be `791.doc.rst`.

To be clear, your file should be `796.doc.rst`. 😁 

That is because how [towncrier](https://pypi.org/project/towncrier/) works, it uses the extension to categorize the changelog entries, so it supports `123.feature`, `456.doc`, and so on. It later added support to have an extra extension that defines the file format (`.md` or `.rst`) to avoid confusion, so we use `123.feature.rst`, `456.doc.rst`, and so on.

> i think this realy needs contributing section

I agree. 👍

Author: well i actualy open that 791.doc i in libre office so yes i was bamboozeled

>     changelog

Reviewer: I think it is best to put the new text in its own file, say `known-limitations.rst`, and add it above `changelog` here. As it stands, it is kind lost at the bottom of the page:

https://pytest-xdist--796.org.readthedocs.build/en/796/

Known limitations
-----------------

> pytest-xdist have some limitation that may be supported in pytest but can't be supported in pytest-xdist

Reviewer: ```suggestion

pytest-xdist has some limitations that may be supported in pytest but can't be supported in pytest-xdist.
```

Author: will fix

> Order and amount of test must be constant

Reviewer: ```suggestion
Order and amount of test must be consistent
```

I think "consistent" makes more sense here, but I'm not a native speaker so I'm not 100% sure.

Author: you are right... Will add it in my next commit

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Is is not possible to have test that differs in order or their amount across threads/execution.

Reviewer: ```suggestion
Is is not possible to have tests that differ in order or their amount across workers.
```

Author: this realy reads better 👍 good sugestion

> This is especially true with `pytest.mark.parametrize`.

Reviewer: ```suggestion
This is especially true with ``pytest.mark.parametrize``.
```

When parametrize is used with set or other unordered iterable-like/generator
pytest-xdist

Example
::
    import pytest

    @pytest.mark.parametrize("param",{"a","b"})
    def test_pytest_parametrize_unordered(param):
        pass

> will cause issue despite pytest supports it.

Reviewer: ```suggestion
In the example above, the fact that ``set``s are not necessarily ordered can cause different workers
to collect tests in different order, which will throw an error.
```

> Truncated pytest signature:

Reviewer: IMHO this last two lines can be removed as I don't think they clarify the problem any better  (a `set` is iterable for example).

I think instead we can add a quick example on how to fix the problem, something like:

```rst
A solution to this is to guarantee that the parametrized values have the same order, for example, call `sorted` on them:

.. code-block:: python

    import pytest

    @pytest.mark.parametrize("param", sorted({"a","b"}))
    def test_pytest_parametrize_unordered(param):
        pass
```

Author: will add another solution with array as  1st example
since it looks to me more tidy and requires change of only 2 characters

it is always best to have multiple options to fix things
